# V3 empirical study — 1-day hourly calibration

**This notebook is the computational source of truth.** The PDF in this folder is written from the same `payload` dict.

| Item | Setting |
|------|---------|
| Models | GBM, GARCH, Heston, Merton, GARCH–Merton, Heston–Merton |
| Companies | SPY, AAPL, MSFT |
| Windows | 12 third-Friday monthly expiries, 21 Oct 2022 – 15 Sep 2023 |
| Evaluation | **Friday immediately before expiry only (hourly 09:59–15:59)** |
| Calibration | **1-day** lookback, **hourly** rolling, 1-minute RTH data |
| LSM horizon | remaining time **to window Friday 15:59**, not listed expiry |
| LSM grid | Δt = **1 minute(s)**, n_steps = remaining RTH / Δt |
| Paths | `10000`, seed `42` |
| Sample | one nearest-ATM listed call per hourly timestamp, DTE 7–60 |

Green / Mark 1 = lowest RMSE%. Friday 15:59 is scheduled but LSM is skipped (`remaining_horizon = 0`).


## Methodology

- Shared sample: nearest-ATM, DTE 7–60, listed expiry, as-of each hourly *t*. All six models price the same contracts.
- No look-ahead: parameters at *t* use only equity returns and listed quotes with timestamp ≤ *t*.
- V3 Heston: option-implied NLS; Euler uses *v_t* then updates variance.
- LSM on the full path cloud. Start every path at the observed *S_t*.


In [1]:
import sys
from pathlib import Path
from IPython.display import display, HTML
ROOT = Path.cwd()
for cand in [ROOT, *ROOT.parents]:
    scripts = cand / 'V3-Models_result' / 'scripts'
    if scripts.exists():
        sys.path.insert(0, str(scripts))
        break
import run_v3_1d_hourly_empirical_study as study
RECOMPUTE = False
payload = study.run_or_load(recompute=RECOMPUTE)
print('cells', len(payload['cells']), 'failures', payload['meta']['failures'])

## Shared sample *n*


In [1]:
display(HTML('shared n'))

Window,Eval Friday,SPY n,AAPL n,MSFT n
Oct 2022 2022-10-21,2022-10-14,6,6,6
Nov 2022 2022-11-18,2022-11-11,6,6,6
Dec 2022 2022-12-16,2022-12-09,6,6,6
Jan 2023 2023-01-20,2023-01-13,6,6,6
Feb 2023 2023-02-17,2023-02-10,6,6,6
Mar 2023 2023-03-17,2023-03-10,6,6,6
Apr 2023 2023-04-21,2023-04-14,6,6,6
May 2023 2023-05-19,2023-05-12,6,6,6
Jun 2023 2023-06-16,2023-06-09,6,6,6
Jul 2023 2023-07-21,2023-07-14,6,6,6


## SPY


### Table SPY-1  ·  Oct 2022  2022-10-21  ·  n = 6  ·  BEST = GARCH–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2022-10-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,41.86,5.9376,-5.9376,-37.80,92.9%,2
GARCH,42.00,5.9640,-5.9640,-37.96,96.2%,4
Heston,42.05,6.0098,-6.0098,-38.25,97.2%,5
Merton,41.96,5.9472,-5.9472,-37.86,95.1%,3
GARCH–Merton,41.39,5.8648,-5.8648,-37.33,90.4%,1
Heston–Merton,42.09,5.9976,-5.9976,-38.18,94.0%,6


### Table SPY-2  ·  Nov 2022  2022-11-18  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2022-11-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,10.0600,-10.0600,-100.00,0.0%,1
GARCH,100.00,10.0600,-10.0600,-100.00,0.0%,2
Heston,100.00,10.0600,-10.0600,-100.00,0.0%,3
Merton,100.00,10.0600,-10.0600,-100.00,0.0%,4
GARCH–Merton,100.00,10.0600,-10.0600,-100.00,0.0%,5
Heston–Merton,100.00,10.0600,-10.0600,-100.00,0.0%,6


### Table SPY-3  ·  Dec 2022  2022-12-16  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2022-12-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,9.0400,-9.0400,-100.00,0.0%,5
GARCH,100.00,9.0400,-9.0400,-100.00,0.0%,4
Heston,99.99,9.0390,-9.0390,-99.99,0.0%,2
Merton,100.00,9.0400,-9.0400,-100.00,0.0%,6
GARCH–Merton,100.00,9.0399,-9.0399,-100.00,0.0%,3
Heston–Merton,99.99,9.0390,-9.0390,-99.99,0.1%,1


### Table SPY-4  ·  Jan 2023  2023-01-20  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-01-20'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,8.1200,-8.1200,-100.00,0.0%,1
GARCH,100.00,8.1200,-8.1200,-100.00,0.0%,2
Heston,100.00,8.1200,-8.1200,-100.00,0.0%,3
Merton,100.00,8.1200,-8.1200,-100.00,0.0%,4
GARCH–Merton,100.00,8.1200,-8.1200,-100.00,0.0%,5
Heston–Merton,100.00,8.1200,-8.1200,-100.00,0.0%,6


### Table SPY-5  ·  Feb 2023  2023-02-17  ·  n = 6  ·  BEST = GARCH–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-02-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,9.1600,-9.1600,-100.00,0.0%,2
GARCH,100.00,9.1600,-9.1600,-100.00,0.0%,3
Heston,100.00,9.1600,-9.1600,-100.00,0.0%,4
Merton,100.00,9.1600,-9.1600,-100.00,0.0%,5
GARCH–Merton,100.00,9.1600,-9.1600,-100.00,0.0%,1
Heston–Merton,100.00,9.1600,-9.1600,-100.00,0.0%,6


### Table SPY-6  ·  Mar 2023  2023-03-17  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-03-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,91.59,8.3781,-8.3781,-91.26,16.7%,4
GARCH,91.83,8.3959,-8.3959,-91.46,16.3%,6
Heston,88.90,8.1107,-8.1107,-88.35,23.2%,2
Merton,91.71,8.3899,-8.3899,-91.39,19.6%,5
GARCH–Merton,91.13,8.3291,-8.3291,-90.73,18.0%,3
Heston–Merton,88.50,8.0720,-8.0720,-87.93,22.6%,1


### Table SPY-7  ·  Apr 2023  2023-04-21  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-04-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.7900,-7.7900,-100.00,0.0%,3
GARCH,100.00,7.7900,-7.7900,-100.00,0.0%,4
Heston,100.00,7.7897,-7.7897,-100.00,0.0%,2
Merton,100.00,7.7900,-7.7900,-100.00,0.0%,5
GARCH–Merton,100.00,7.7900,-7.7900,-100.00,0.0%,6
Heston–Merton,99.99,7.7896,-7.7896,-99.99,0.0%,1


### Table SPY-8  ·  May 2023  2023-05-19  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-05-19'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,6.7800,-6.7800,-100.00,0.0%,1
GARCH,100.00,6.7800,-6.7800,-100.00,0.0%,2
Heston,100.00,6.7800,-6.7800,-100.00,0.0%,3
Merton,100.00,6.7800,-6.7800,-100.00,0.0%,4
GARCH–Merton,100.00,6.7800,-6.7800,-100.00,0.0%,5
Heston–Merton,100.00,6.7800,-6.7800,-100.00,0.0%,6


### Table SPY-9  ·  Jun 2023  2023-06-16  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-06-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,5.2200,-5.2200,-100.00,0.0%,1
GARCH,100.00,5.2200,-5.2200,-100.00,0.0%,2
Heston,100.00,5.2200,-5.2200,-100.00,0.0%,3
Merton,100.00,5.2200,-5.2200,-100.00,0.0%,4
GARCH–Merton,100.00,5.2200,-5.2200,-100.00,0.0%,5
Heston–Merton,100.00,5.2200,-5.2200,-100.00,0.0%,6


### Table SPY-10  ·  Jul 2023  2023-07-21  ·  n = 6  ·  BEST = GARCH–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-07-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,6.4700,-6.4700,-100.00,0.0%,2
GARCH,100.00,6.4700,-6.4700,-100.00,0.0%,3
Heston,100.00,6.4700,-6.4700,-100.00,0.0%,4
Merton,100.00,6.4700,-6.4700,-100.00,0.0%,5
GARCH–Merton,100.00,6.4699,-6.4699,-100.00,0.0%,1
Heston–Merton,100.00,6.4700,-6.4700,-100.00,0.0%,6


### Table SPY-11  ·  Aug 2023  2023-08-18  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-08-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,7.2300,-7.2300,-100.00,0.0%,1
GARCH,100.00,7.2300,-7.2300,-100.00,0.0%,2
Heston,100.00,7.2300,-7.2300,-100.00,0.0%,3
Merton,100.00,7.2300,-7.2300,-100.00,0.0%,4
GARCH–Merton,100.00,7.2300,-7.2300,-100.00,0.0%,5
Heston–Merton,100.00,7.2300,-7.2300,-100.00,0.0%,6


### Table SPY-12  ·  Sep 2023  2023-09-15  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['SPY|2023-09-15'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,5.8200,-5.8200,-100.00,0.0%,1
GARCH,100.00,5.8200,-5.8200,-100.00,0.0%,2
Heston,100.00,5.8200,-5.8200,-100.00,0.0%,3
Merton,100.00,5.8200,-5.8200,-100.00,0.0%,4
GARCH–Merton,100.00,5.8200,-5.8200,-100.00,0.0%,5
Heston–Merton,100.00,5.8200,-5.8200,-100.00,0.0%,6


## AAPL


### Table AAPL-1  ·  Oct 2022  2022-10-21  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2022-10-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,72.72,4.8819,-4.8819,-70.75,56.7%,5
GARCH,73.80,4.9551,-4.9551,-71.81,59.7%,6
Heston,70.82,4.7682,-4.7682,-69.10,53.7%,1
Merton,72.69,4.8751,-4.8751,-70.65,55.0%,4
GARCH–Merton,72.34,4.8362,-4.8362,-70.09,57.2%,3
Heston–Merton,71.44,4.7968,-4.7968,-69.52,55.6%,2


### Table AAPL-2  ·  Nov 2022  2022-11-18  ·  n = 6  ·  BEST = GARCH


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2022-11-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,99.99,5.3496,-5.3496,-99.99,0.1%,5
GARCH,99.86,5.3428,-5.3428,-99.86,0.4%,1
Heston,99.96,5.3479,-5.3479,-99.96,0.3%,4
Merton,99.99,5.3496,-5.3496,-99.99,0.1%,6
GARCH–Merton,99.87,5.3428,-5.3428,-99.87,0.5%,2
Heston–Merton,99.95,5.3476,-5.3476,-99.95,0.3%,3


### Table AAPL-3  ·  Dec 2022  2022-12-16  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2022-12-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,82.45,4.3228,-4.3228,-81.95,41.9%,4
GARCH,82.76,4.3341,-4.3341,-82.16,41.2%,6
Heston,75.58,3.9557,-3.9557,-74.99,41.3%,2
Merton,82.53,4.3271,-4.3271,-82.03,43.1%,5
GARCH–Merton,81.66,4.2730,-4.2730,-81.00,39.6%,3
Heston–Merton,75.55,3.9528,-3.9528,-74.93,37.4%,1


### Table AAPL-4  ·  Jan 2023  2023-01-20  ·  n = 6  ·  BEST = GARCH–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-01-20'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,99.83,5.6655,-5.6655,-99.83,0.9%,5
GARCH,99.67,5.6560,-5.6560,-99.67,1.4%,2
Heston,99.71,5.6584,-5.6584,-99.71,1.6%,3
Merton,99.84,5.6657,-5.6657,-99.84,1.1%,6
GARCH–Merton,99.60,5.6521,-5.6521,-99.60,1.5%,1
Heston–Merton,99.75,5.6609,-5.6609,-99.75,1.4%,4


### Table AAPL-5  ·  Feb 2023  2023-02-17  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-02-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,98.38,5.6321,-5.6321,-98.38,7.0%,4
GARCH,98.73,5.6519,-5.6519,-98.72,4.9%,6
Heston,94.91,5.4318,-5.4318,-94.88,12.8%,2
Merton,98.30,5.6273,-5.6273,-98.29,6.3%,3
GARCH–Merton,98.55,5.6418,-5.6418,-98.55,5.5%,5
Heston–Merton,94.76,5.4227,-5.4227,-94.72,11.8%,1


### Table AAPL-6  ·  Mar 2023  2023-03-17  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-03-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,84.86,3.8935,-3.8935,-84.18,31.9%,5
GARCH,83.73,3.8332,-3.8332,-82.88,29.7%,4
Heston,78.28,3.5671,-3.5671,-77.13,33.8%,1
Merton,84.94,3.8979,-3.8979,-84.28,33.2%,6
GARCH–Merton,83.38,3.8154,-3.8154,-82.50,31.2%,3
Heston–Merton,78.31,3.5685,-3.5685,-77.16,32.9%,2


### Table AAPL-7  ·  Apr 2023  2023-04-21  ·  n = 6  ·  BEST = Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-04-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,96.51,5.3038,-5.3038,-96.43,13.7%,3
GARCH,97.51,5.3608,-5.3608,-97.47,11.7%,6
Heston,96.58,5.3071,-5.3071,-96.49,14.8%,4
Merton,96.50,5.3037,-5.3037,-96.43,13.9%,1
GARCH–Merton,97.14,5.3398,-5.3398,-97.09,12.4%,5
Heston–Merton,96.50,5.3031,-5.3031,-96.42,13.2%,2


### Table AAPL-8  ·  May 2023  2023-05-19  ·  n = 6  ·  BEST = GARCH–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-05-19'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,4.2500,-4.2500,-100.00,0.0%,3
GARCH,100.00,4.2500,-4.2500,-100.00,0.0%,4
Heston,100.00,4.2500,-4.2500,-100.00,0.0%,2
Merton,100.00,4.2500,-4.2500,-100.00,0.0%,5
GARCH–Merton,100.00,4.2500,-4.2500,-100.00,0.0%,1
Heston–Merton,100.00,4.2500,-4.2500,-100.00,0.0%,6


### Table AAPL-9  ·  Jun 2023  2023-06-16  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-06-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,3.4500,-3.4500,-100.00,0.0%,5
GARCH,100.00,3.4500,-3.4500,-100.00,0.0%,3
Heston,99.93,3.4477,-3.4477,-99.93,0.3%,2
Merton,100.00,3.4500,-3.4500,-100.00,0.0%,6
GARCH–Merton,100.00,3.4500,-3.4500,-100.00,0.0%,4
Heston–Merton,99.93,3.4475,-3.4475,-99.93,0.2%,1


### Table AAPL-10  ·  Jul 2023  2023-07-21  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-07-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,5.5500,-5.5500,-100.00,0.0%,1
GARCH,100.00,5.5500,-5.5500,-100.00,0.0%,2
Heston,100.00,5.5500,-5.5500,-100.00,0.0%,3
Merton,100.00,5.5500,-5.5500,-100.00,0.0%,4
GARCH–Merton,100.00,5.5500,-5.5500,-100.00,0.0%,5
Heston–Merton,100.00,5.5500,-5.5500,-100.00,0.0%,6


### Table AAPL-11  ·  Aug 2023  2023-08-18  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-08-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,3.4000,-3.4000,-100.00,0.0%,1
GARCH,100.00,3.4000,-3.4000,-100.00,0.0%,2
Heston,100.00,3.4000,-3.4000,-100.00,0.0%,3
Merton,100.00,3.4000,-3.4000,-100.00,0.0%,4
GARCH–Merton,100.00,3.4000,-3.4000,-100.00,0.0%,5
Heston–Merton,100.00,3.4000,-3.4000,-100.00,0.0%,6


### Table AAPL-12  ·  Sep 2023  2023-09-15  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['AAPL|2023-09-15'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,3.8250,-3.8250,-100.00,0.0%,6
GARCH,100.00,3.8249,-3.8249,-100.00,0.0%,4
Heston,99.96,3.8234,-3.8234,-99.96,0.2%,1
Merton,100.00,3.8250,-3.8250,-100.00,0.0%,5
GARCH–Merton,99.99,3.8246,-3.8246,-99.99,0.0%,3
Heston–Merton,99.96,3.8235,-3.8235,-99.96,0.2%,2


## MSFT


### Table MSFT-1  ·  Oct 2022  2022-10-21  ·  n = 6  ·  BEST = GARCH–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2022-10-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,44.10,4.4067,-4.4067,-41.28,85.2%,6
GARCH,43.95,4.4027,-4.4027,-41.24,91.2%,4
Heston,43.89,4.4108,-4.4108,-41.32,83.0%,2
Merton,44.01,4.3879,-4.3879,-41.10,88.2%,5
GARCH–Merton,43.52,4.3457,-4.3457,-40.71,87.3%,1
Heston–Merton,43.93,4.4167,-4.4167,-41.37,83.7%,3


### Table MSFT-2  ·  Nov 2022  2022-11-18  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2022-11-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,96.77,5.9991,-5.9991,-96.76,8.5%,4
GARCH,97.47,6.0424,-6.0424,-97.46,7.0%,6
Heston,93.95,5.8233,-5.8233,-93.92,11.8%,1
Merton,96.66,5.9920,-5.9920,-96.64,9.2%,3
GARCH–Merton,97.09,6.0192,-6.0192,-97.08,8.1%,5
Heston–Merton,93.95,5.8234,-5.8234,-93.93,11.7%,2


### Table MSFT-3  ·  Dec 2022  2022-12-16  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2022-12-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,22.48,1.3291,-1.3291,-19.99,94.1%,4
GARCH,23.49,1.3765,-1.3765,-20.70,96.7%,6
Heston,17.27,1.0070,-1.0070,-15.14,76.4%,1
Merton,22.40,1.3243,-1.3243,-19.92,94.1%,3
GARCH–Merton,22.99,1.3550,-1.3550,-20.38,95.8%,5
Heston–Merton,17.67,1.0120,-1.0120,-15.22,78.7%,2


### Table MSFT-4  ·  Jan 2023  2023-01-20  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-01-20'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,24.91,1.7825,-1.7825,-22.63,95.6%,3
GARCH,25.10,1.7959,-1.7959,-22.80,96.4%,6
Heston,24.87,1.7725,-1.7725,-22.51,89.1%,2
Merton,25.03,1.7863,-1.7863,-22.68,95.4%,5
GARCH–Merton,24.98,1.7863,-1.7863,-22.68,98.2%,4
Heston–Merton,24.85,1.7807,-1.7807,-22.61,92.8%,1


### Table MSFT-5  ·  Feb 2023  2023-02-17  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-02-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,99.29,7.3470,-7.3470,-99.28,2.3%,5
GARCH,99.10,7.3327,-7.3327,-99.09,2.3%,4
Heston,96.73,7.1552,-7.1552,-96.69,5.7%,1
Merton,99.31,7.3485,-7.3485,-99.30,2.2%,6
GARCH–Merton,99.09,7.3324,-7.3324,-99.09,3.0%,3
Heston–Merton,96.83,7.1629,-7.1629,-96.80,6.0%,2


### Table MSFT-6  ·  Mar 2023  2023-03-17  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-03-17'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,95.62,4.9643,-4.9643,-95.47,7.4%,6
GARCH,94.94,4.9232,-4.9232,-94.68,6.9%,4
Heston,89.14,4.6065,-4.6065,-88.59,12.6%,2
Merton,95.60,4.9630,-4.9630,-95.44,8.2%,5
GARCH–Merton,94.45,4.8967,-4.8967,-94.17,7.3%,3
Heston–Merton,88.96,4.5948,-4.5948,-88.36,12.7%,1


### Table MSFT-7  ·  Apr 2023  2023-04-21  ·  n = 6  ·  BEST = Heston–Merton


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-04-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,98.40,9.7146,-9.7146,-98.38,7.4%,3
GARCH,99.24,9.7990,-9.7990,-99.23,4.8%,6
Heston,97.33,9.6064,-9.6064,-97.28,11.4%,2
Merton,98.42,9.7170,-9.7170,-98.40,8.7%,4
GARCH–Merton,98.77,9.7518,-9.7518,-98.75,6.7%,5
Heston–Merton,97.24,9.5966,-9.5966,-97.18,10.1%,1


### Table MSFT-8  ·  May 2023  2023-05-19  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-05-19'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,8.7750,-8.7750,-100.00,0.0%,1
GARCH,100.00,8.7750,-8.7750,-100.00,0.0%,2
Heston,100.00,8.7750,-8.7750,-100.00,0.0%,3
Merton,100.00,8.7750,-8.7750,-100.00,0.0%,4
GARCH–Merton,100.00,8.7750,-8.7750,-100.00,0.0%,5
Heston–Merton,100.00,8.7750,-8.7750,-100.00,0.0%,6


### Table MSFT-9  ·  Jun 2023  2023-06-16  ·  n = 6  ·  BEST = Heston


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-06-16'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,6.8000,-6.8000,-100.00,0.0%,3
GARCH,100.00,6.8000,-6.8000,-100.00,0.0%,4
Heston,100.00,6.7998,-6.7998,-100.00,0.0%,1
Merton,100.00,6.8000,-6.8000,-100.00,0.0%,5
GARCH–Merton,100.00,6.8000,-6.8000,-100.00,0.0%,6
Heston–Merton,100.00,6.7999,-6.7999,-100.00,0.0%,2


### Table MSFT-10  ·  Jul 2023  2023-07-21  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-07-21'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,10.4000,-10.4000,-100.00,0.0%,1
GARCH,100.00,10.4000,-10.4000,-100.00,0.0%,2
Heston,100.00,10.4000,-10.4000,-100.00,0.0%,3
Merton,100.00,10.4000,-10.4000,-100.00,0.0%,4
GARCH–Merton,100.00,10.4000,-10.4000,-100.00,0.0%,5
Heston–Merton,100.00,10.4000,-10.4000,-100.00,0.0%,6


### Table MSFT-11  ·  Aug 2023  2023-08-18  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-08-18'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,6.9500,-6.9500,-100.00,0.0%,1
GARCH,100.00,6.9500,-6.9500,-100.00,0.0%,2
Heston,100.00,6.9500,-6.9500,-100.00,0.0%,3
Merton,100.00,6.9500,-6.9500,-100.00,0.0%,4
GARCH–Merton,100.00,6.9500,-6.9500,-100.00,0.0%,5
Heston–Merton,100.00,6.9500,-6.9500,-100.00,0.0%,6


### Table MSFT-12  ·  Sep 2023  2023-09-15  ·  n = 6  ·  BEST = GBM


In [1]:
display(HTML(study.emp._html_table(payload['tables']['MSFT|2023-09-15'])))

Model,RMSE%,MAE,Bias $,Bias%,Early-ex. frac.,Mark
GBM,100.00,9.6000,-9.6000,-100.00,0.0%,1
GARCH,100.00,9.6000,-9.6000,-100.00,0.0%,2
Heston,100.00,9.6000,-9.6000,-100.00,0.0%,3
Merton,100.00,9.6000,-9.6000,-100.00,0.0%,4
GARCH–Merton,100.00,9.6000,-9.6000,-100.00,0.0%,5
Heston–Merton,100.00,9.6000,-9.6000,-100.00,0.0%,6


## Summary S1


In [1]:
display(HTML('s1'))

Rank,Model,Mean RMSE%,Median RMSE%,# of 36 cells best
1,Heston,89.16,99.97,7
2,Heston–Merton,89.17,99.97,9
3,GARCH–Merton,90.17,100.00,6
4,GBM,90.27,100.00,12
5,Merton,90.27,100.00,1
6,GARCH,90.37,100.00,1


## Findings

- On mean percentage RMSE across the 36 company×window cells, Heston is the best overall model (mean RMSE% = 89.16; 7 of 36 cells).

- Second overall is Heston–Merton (mean RMSE% = 89.17; 9 cells).

- The ranking is not stable. Best-in-cell models across the 36 tables: GARCH, GARCH–Merton, GBM, Heston, Heston–Merton, Merton.

- SPY: lowest mean RMSE% is Heston–Merton (94.21); wins 3 of 12 windows.

- AAPL: lowest mean RMSE% is Heston (92.98); wins 3 of 12 windows.

- MSFT: lowest mean RMSE% is Heston (80.26); wins 4 of 12 windows.

- Jump models as a group have lower mean RMSE% (89.87) than the three no-jump models (89.93).

- These conclusions are conditional on the V3 filters, 1-day hourly calibration, nearest-ATM hourly sample, LSM with 10000 paths and 1-minute steps to Friday 15:59, and American-call quotes only. They are not a statement about European options or other tenors.
